File: 백준/Silver/1629. 곱셈/곱셈.js
const [A, B, C] = require("fs")
  .readFileSync(0)
  .toString()
  .trim()
  .split(" ")
  .map(BigInt);
function pow(b) {
  if (b === 1n) return A % C;
  const temp = pow(b / 2n);
  if (b % 2n === 0n) return (temp * temp) % C;
  else return (temp * temp * A) % C;
}

console.log(Number(pow(B)));

File: 백준/Silver/1629. 곱셈/곱셈.py
a,b,c = map(int, input().split())

def recursion(a,b,c):
    if b==1:
        return a%c
    if b%2==0:
        return (recursion(a,b//2,c)**2)%c
    else:
        return (recursion(a,b//2,c)**2)*a%c
print(recursion(a,b,c))